langserver/hover.go

```diff
@@ -324,8 +407,39 @@ func (h *LangHandler) handleHoverGodef(ctx context.Context, conn jsonrpc2.JSONRP
 	}
 
 	loc := goRangeToLSPLocation(fset, res.Start, res.End)
-
 	if loc.URI == "file://" {
+		_, node, _, _, pkg, _, err := h.typecheck(ctx, conn, params.TextDocument.URI, params.Position)
+		if err != nil {
+			// Invalid nodes means we tried to click on something which is
+			// not an ident (eg comment/string/etc). Return no information.
+			if _, ok := err.(*invalidNodeError); ok {
+				return nil, nil
```

Comment:
what is the difference between returning `nil, nil` here and `&lsp.Hover{}, nil` on L444?

Reply:
`&lsp.Hover{}, nil` is returned to indicate that there is a valid identifier being hovered that isn't currently supported (builtins). For invalid identifiers (such as `{`, `=`, `func`, etc), we return `nil, nil`. From the user's perspective, the difference is unnoticeable. 

I've removed the `&lsp.Hover{}, nil` in the next commit.